### PR TITLE
CallerArgumentExpression: Warning for self-referential

### PR DESCRIPTION
**Pull request by Youssef1313**
Proposal: dotnet/csharplang#287
Test plan: https://github.com/dotnet/roslyn/issues/52745

Addresses part of LDM notes: https://github.com/dotnet/csharplang/blob/main/meetings/2021/LDM-2021-06-14.md

### CODE REVIEW

**Comment by Youssef1313** [comment]
CI is stuck. Closing & reopening.

File: src/Compilers/CSharp/Portable/CSharpResources.resx
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/CSharpResources.resx:6668]: >The CallerArgumentExpressionAttribute is applied with a self-referential parameter. [](http://example.com/codeflow?start=11&length=83)

We usually use the original message with removed placeholders. i think this will work well in this case as well: "The CallerArgumentExpressionAttribute applied to parameter will have no effect because it's self-referential."

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs:411]: >.VerifyDiagnostics( [](http://example.com/codeflow?start=23&length=19)

Consider calling this on the result returned by ```CompileAndVerify``` instead.

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs:391]: >TestArgumentExpressionIsSelfReferential [](http://example.com/codeflow?start=20&length=39)

Do we have a test covering consumption of similar attribute from metadata? Perhaps we can compile similar class to image reference, confirm the attribute is there and test the same scenario against that image.

**Comment by AlekseyTs** [comment]
Done with the review pass (commit 1)

**Comment by Youssef1313** [comment]
@AlekseyTs Thanks for review. I addressed your feedback.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 3)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler, @333fred Please review, need a second review for a Community PR.

**Review by 333fred — APPROVED** [review]
LGTM (commit 3)

**Comment by 333fred** [comment]
Thanks @Youssef1313!

**Pull request merged**
